processing/src/main/java/org/apache/druid/query/filter/ArrayContainsFilter.java

```diff
@@ -0,0 +1,677 @@
+  {
+    return this;
+  }
+
+  @JsonProperty
+  public String getColumn()
+  {
+    return column;
+  }
+
+  @JsonProperty
+  public ColumnType getElementMatchValueType()
+  {
+    return elementMatchValueType;
+  }
+
+  @JsonProperty
+  public Object getElementMatchValue()
+  {
+    return elementMatchValue;
+  }
+
+  @Nullable
+  @JsonProperty
+  @JsonInclude(JsonInclude.Include.NON_NULL)
+  public FilterTuning getFilterTuning()
+  {
+    return filterTuning;
+  }
+
+  @Override
+  public String toString()
+  {
+    DimFilter.DimFilterToStringBuilder bob =
+        new DimFilter.DimFilterToStringBuilder().append("array_contains(")
+                                                .appendDimension(column, null)
+                                                .append(", ")
+                                                .append(elementMatchValueEval.value())
+                                                .append(")");
+    if (!ColumnType.STRING.equals(elementMatchValueType)) {
+      bob.append(" (" + elementMatchValueType.asTypeString() + ")");
+    }
+    return bob.appendFilterTuning(filterTuning).build();
+  }
+
+  @Override
+  public boolean equals(Object o)
+  {
+    if (this == o) {
+      return true;
+    }
+    if (o == null || getClass() != o.getClass()) {
+      return false;
+    }
+    ArrayContainsFilter that = (ArrayContainsFilter) o;
+    if (!column.equals(that.column)) {
+      return false;
+    }
+    if (!Objects.equals(elementMatchValueType, that.elementMatchValueType)) {
+      return false;
+    }
+    if (!Objects.equals(filterTuning, that.filterTuning)) {
+      return false;
+    }
+    if (elementMatchValueType.isArray()) {
+      return Arrays.deepEquals(elementMatchValueEval.asArray(), that.elementMatchValueEval.asArray());
+    } else {
+      return Objects.equals(elementMatchValueEval.value(), that.elementMatchValueEval.value());
+    }
+  }
+
+  @Override
+  public int hashCode()
+  {
+    return Objects.hash(column, elementMatchValueType, elementMatchValueEval.value(), filterTuning);
+  }
+
+  @Override
+  public RangeSet<String> getDimensionRangeSet(String dimension)
+  {
+    return null;
+  }
+
+  @Nullable
+  @Override
+  public BitmapColumnIndex getBitmapColumnIndex(ColumnIndexSelector selector)
+  {
+    if (!Filters.checkFilterTuningUseIndex(column, selector, filterTuning)) {
+      return null;
+    }
+
+    final ColumnIndexSupplier indexSupplier = selector.getIndexSupplier(column);
+    if (indexSupplier == null) {
+      return new AllUnknownBitmapColumnIndex(selector);
+    }
+    final ArrayElementIndexes elementIndexes = indexSupplier.as(ArrayElementIndexes.class);
+    if (elementIndexes != null) {
+      return elementIndexes.containsValue(elementMatchValueEval.value(), elementMatchValueType);
+    }
+
+    if (selector.getColumnCapabilities(column) != null && !selector.getColumnCapabilities(column).isArray()) {
+      // column is not an array, behave like a normal equality filter
```

Comment:
Looks like some copypaste from `EqualityFilter`, would it make sense to delegate to that or extract a helper method?

Reply:
yea, will work on consolidating some of this stuff